ftw/mobile/buttons.py

```diff
@@ -99,3 +102,31 @@ def endpoint(self):
     def startup_cachekey(self):
         return (self.context.restrictedTraverse('@@mobilenav')
                 .get_startup_cachekey())
+
+class MultilanguageButton(BaseButton):
+
+    def label(self):
+        context = self.context.aq_inner
+        portal_state = getMultiAdapter((context, self.request), name=u'plone_portal_state')
+        current_language = portal_state.language()
+        return current_language
+
+    def position(self):
+        return 300
+
+    def data(self):
+        result = OrderedDict()
+        portal_languages = self.context.portal_languages
+
+        langs = portal_languages.getAvailableLanguages()
```

Comment:
I'm not sure if this is the "plone.app.multilingual" approach and I have no idea how to do this correctly.
I assume it should not be a "switch plone GUI language" switch but a "switch to a different language version of this context" switch.

@maethu can you give your input how this should be done?  I have no idea how this should work.